tests/ui/min_ident_chars.rs

```diff
@@ -0,0 +1,81 @@
+//@aux-build:proc_macros.rs
+#![allow(irrefutable_let_patterns, nonstandard_style, unused)]
+#![warn(clippy::min_ident_chars)]
+
+extern crate proc_macros;
+use proc_macros::external;
+use proc_macros::with_span;
+
+struct A {
+    a: u32,
+    i: u32,
+    A: u32,
+    I: u32,
+}
+
+struct B(u32);
+
+struct O {
+    o: u32,
+}
+
+struct i;
+
+enum C {
+    D,
+    E,
+    F,
+    j,
+}
+
+struct Vec4 {
+    x: u32,
+    y: u32,
+    z: u32,
+    w: u32,
+}
+
+struct AA<T, E>(T, E);
+
+fn main() {
```

Comment:
Could you add a test for function parameters like this:

```rs
// Should lint:
fn wrong_pythagoras(a: f32, b: f32) -> f32 {
    a * a + a * b
}
```

and one for just the underscore identifier:

```rs
// Should not lint
for _ in 0..10 {}
```

The later one could maybe lint, but then we should add `_` as a default allowed ident

Reply:
Done, the latter was already handled for by only checking `Binding`s (that will be a `Wild`, afaik)